jerry-core/api/jerry.c

```diff
@@ -3025,6 +3025,64 @@ jerry_get_property_by_index (const jerry_value_t obj_val, /**< object value */
   return jerry_return (ret_value);
 } /* jerry_get_property_by_index */
 
+/**
+ * Get the own property value of an object with the given name.
+ *
+ * Note:
+ *      returned value must be freed with jerry_release_value, when it is no longer needed.
+ *
+ * @return value of the property - if success
+ *         value marked with error flag - otherwise
+ */
+jerry_value_t
+jerry_get_own_property (const jerry_value_t obj_val, /**< object value */
+                        const jerry_value_t prop_name_val, /**< property name (string value) */
+                        const jerry_value_t receiver_val, /**< receiver object value */
+                        bool *found_p) /**< [out] true, if the property is found, false otherwise */
+{
+  jerry_assert_api_available ();
+
+  if (!ecma_is_value_object (obj_val)
+      || !ecma_is_value_prop_name (prop_name_val)
+      || !ecma_is_value_object (receiver_val))
+  {
+    return jerry_throw (ecma_raise_type_error (ECMA_ERR_MSG (wrong_args_msg_p)));
+  }
+
+  ecma_object_t *object_p = ecma_get_object_from_value (obj_val);
+  ecma_string_t *property_name_p = ecma_get_prop_name_from_value (prop_name_val);
+
+#if JERRY_BUILTIN_PROXY
+  if (ECMA_OBJECT_IS_PROXY (object_p))
+  {
+    if (found_p != NULL)
+    {
+      *found_p = true;
```

Comment:
AFAIK in case of proxy it is not guaranteed if the property is found. Eg.:
```
var data = new Proxy({}, {});
``` 
for this `data` this method will return true for all property. Do we want this?

Reply:
Yes. If special handling is needed, the user can do it in the API. Another option is never returning anything with Proxy, but I think this is better.